src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/RadioactivityTask.java

```diff
@@ -0,0 +1,150 @@
+        }
+    }
+
+    private void handleRadiation(@Nonnull Player p, @Nonnull PlayerProfile profile) {
+        if (p.getGameMode() == GameMode.CREATIVE || p.getGameMode() == GameMode.SPECTATOR) {
+            return;
+        }
+
+        int exposureTotal = 0;
+        UUID uuid = p.getUniqueId();
+
+        if (!profile.hasFullProtectionAgainst(ProtectionType.RADIATION)) {
+            for (ItemStack item : p.getInventory()) {
+                if (item == null || item.getType().isAir()) {
+                    continue;
+                }
+                SlimefunItem sfItem = SlimefunItem.getByItem(item);
+                if (sfItem instanceof RadioactiveItem) {
+                    exposureTotal += item.getAmount() * ((RadioactiveItem) sfItem).getRadioactivity().getExposureModifier();
+                }
+            }
+        }
+        
+        int exposureLevelBefore = radioactivityLevel.getOrDefault(uuid, 0);
+        if (exposureTotal > 0) {
+            if (exposureLevelBefore == 0) {
+                SlimefunPlugin.getLocalization().sendMessage(p, "messages.radiation");
+            }
+            radioactivityLevel.put(uuid, Math.min(exposureLevelBefore + exposureTotal, 100));
+        } else if (exposureLevelBefore > 0) {
+            radioactivityLevel.put(uuid, exposureLevelBefore - 1);
+        }
+        
+        int exposureLevelAfter = radioactivityLevel.getOrDefault(uuid, 0);
+        for (Symptom symptom : SYMPTOMS) {
+            if (symptom.minExposure <= exposureLevelAfter) {
+                applySymptom(symptom, p);
+            }
+        }
+        
+        if (exposureLevelAfter > 0 || exposureLevelBefore > 0) {
+            String msg = SlimefunPlugin.getLocalization().getMessage(p, "actionbar.radiation")
+                    .replace("%level%", "" + exposureLevelAfter);
+            p.spigot().sendMessage(ChatMessageType.ACTION_BAR,
+                    new ComponentBuilder().append(ChatColors.color(msg)).create()
+            );
+        }
+    }
+
+    /**
+     * Applies a symptom to the player.
+     *
+     * @param s Symptom to apply
+     * @param p Player to apply to
+     */
+    private void applySymptom(@Nonnull Symptom s, @Nonnull Player p) {
+        SlimefunPlugin.runSync(() -> {
+            switch (s) {
+                case SLOW: {
+                    p.addPotionEffect(SLOW);
+                    break;
+                }
+                case WITHER_LOW: {
+                    p.addPotionEffect(WITHER);
+                    break;
+                }
+                case BLINDNESS: {
+                    p.addPotionEffect(BLINDNESS);
+                    break;
+                }
+                case WITHER_HIGH: {
+                    p.addPotionEffect(WITHER2);
+                    break;
+                }
+                case IMMINENT_DEATH: {
+                    p.setHealth(0);
+                    break;
+                }
```

Comment:
1. No braces when using a switch/clause.
2. This is an enum, you don't need to use a switch/case for enums, you can just attach a `Consumer<Player>` to each enum constant for this.

Reply:
Where should I place the enum btw?